pkg/config/store/fsstore2.go

```diff
@@ -0,0 +1,184 @@
+// Copyright 2017 Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package store
+
+import (
+	"context"
+	"crypto/sha1"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/ghodss/yaml"
+	"github.com/golang/glog"
+)
+
+const defaultDuration = time.Second / 2
+
+// fsStore2 is Store2Backend implementation using filesystem.
+type fsStore2 struct {
+	root          string
+	kinds         []string
+	checkDuration time.Duration
+	chs           *ContextChList
+
+	mu   sync.RWMutex
+	data map[Key]map[string]interface{}
+	shas map[Key][sha1.Size]byte
+}
+
+var _ Store2Backend = &fsStore2{}
+
+func (s *fsStore2) readFiles() (map[Key][]byte, map[Key][sha1.Size]byte) {
+	const suffix = ".yaml"
+	result := map[Key][]byte{}
+
+	for _, kind := range s.kinds {
+		root := filepath.Join(s.root, kind)
+		err := filepath.Walk(root, func(path string, info os.FileInfo, err error) error {
+			if err != nil {
+				return err
+			}
+			if strings.HasSuffix(path, suffix) && (info.Mode()&os.ModeType) == 0 {
+				data, err := ioutil.ReadFile(path)
```

Comment:
I see what you are doing here, however what we really want is to consume the same artifacts that we will send via kubectl create -f

1. each yaml file with `---` as a separator. 
2. Each element is full k8s doc.
```
apiVersion: config.istio.io/v1alpha2
kind: Metric
metadata:
  name: requestcount
  namespace: default
spec:
  description: request count by source, target, service, and code
  kind: COUNTER
  labels:
    method: 1
    response_code: 2
    service: 1
    source: 1
    target: 1
    version: 1
  value: INT64
``` 
This way we can test with the same files.